### ISSUE TITLE
bug with alpha and render order

### ISSUE THREAD

**Issue by sg3des** [tbogdala/fizzle#7]
example - https://github.com/sg3des/fizzle-example

Found a problem - if object with transparency material render before other objects, transparent will not.
in example look at 219-221 lines and try swap objects.

add screenshots, [transparent object first](https://github.com/sg3des/fizzle-example/blob/master/smoke%20before.png), [last](https://github.com/sg3des/fizzle-example/blob/master/smoke%20after.png)

**Comment by tbogdala** [comment]
Yes, as far as I understand it, that's "normal" behavior. I believe the common solution to this is to render transparent objects in a last pass.

I haven't set up any built-in ways of doing this yet. It sounds like something a _scene manager_ type of object would organize -- what objects to draw and when to draw them. Something like this goes on in the lights_forward sample in that it does a shadow pass first for the lights and then does the actual render. The third step would be to render transparencies.

I'm getting to the point where I have to make a decision on this in my game prototypes. I think I'll end up making a basic naive scene manager and an interface for it. It will also require a flag on renderables indicating if they're transparent or not.

Soon.

**Comment by sg3des** [comment]
Hello, I use this solution:
1) enable next opengl flags:

```
gfx.Enable(graphicsprovider.BLEND)
gfx.BlendFunc(graphicsprovider.SRC_ALPHA, graphicsprovider.ONE_MINUS_SRC_ALPHA)
gfx.Enable(graphicsprovider.SAMPLE_ALPHA_TO_COVERAGE)
gfx.Enable(graphicsprovider.EQUAL)
```

2) all objects have flag `Transparency` 
3) render objects in two steps, first  `transparancy = false`, second `transparency = true`.

Result is not good, but acceptable.

**Comment by tbogdala** [comment]
Yea, something like that is what I was thinking. Thanks for posting your solution.
